### PR TITLE
Resolve 156: Subgroups overlap when scrolling vertically

### PR DESCRIPTION
**Pull request by jczacharia**
**Problem:** 

This line (in Group.js)

```
stack.nostack(this.visibleItems, margin, this.subgroups, this.itemSet.options.stackSubgroups);
```
in the function `_redrawItems` uses the `this.subgroups` `top` and `height` properties to set its items `top` property.

Since `_updateSubGroupHeights` is called before `this.visibleItems` is set in `_redrawItems`, then the subgroups height and visible are 0 and false accordingly which sets the items in the 2nd+ subgroups items top property to be 0 and thus overlap all the items in the group.

**Soution:**

Call `_updateSubGroupHeights` after setting `this.visibleItems` but before `stack.nostack(...`

I took the function `_updateSubGroupHeights` out of the redraw queue and called it each time `this.visibleItems` was set in `_redrawItems`.

This seems to fix the problem but I am not sure if this will cause any unwanted side-effects since `_updateSubGroupHeights` is not guaranteed to be called each group redraw.

Any suggestions on improvements would be appreciated!

Demo:
defect_fix_subgroupoverlay

### CODE REVIEW

File: lib/timeline/component/Group.js [hunk 1]
```diff
@@ -503,9 +506,6 @@ class Group {
         forceRestack = this._didMarkerHeightChange.call(this) || forceRestack;
       },
 
-      // recalculate the height of the subgroups
-      this._updateSubGroupHeights.bind(this, margin),
```
Comment (yotamberk) [inline lib/timeline/component/Group.js]: Removing this isn't necessary. Adding the above should have done the trick.

Comment (jczacharia) [reply lib/timeline/component/Group.js]: I removed it just to avoid calling it twice. But I can add it back in.

Comment (yotamberk) [reply lib/timeline/component/Group.js]: I think you should. Just to avoid special cases

Comment (yotamberk) [reply lib/timeline/component/Group.js]: I think you should. Just to avoid special cases

Comment (yotamberk) [reply lib/timeline/component/Group.js]: I think you should. Just to avoid special cases

**Pull request merged**